### PR TITLE
Added style guide and Python 3 guide

### PR DESCRIPTION
**Pull request by Onager**

### CODE REVIEW

File: process/Style-guide.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+Log2Timeline's style is based on the the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html), with specific 
+variations and expansion spelled out in the sections below.
+
+## Compatibility
+All new log2timeline code should be compatible with both Python 3.4+ and Python 2.7+. The [Python 3 Guide](Python-3-Guide) has some more detail about compatibility issues, and the pylint configuration will also flag some issues.
+
+
+## Indentation
+
+* Indent your code blocks with 2 spaces (not 4 as in the style guide).
```
Comment (joachimmetz) [inline process/Style-guide.md]: "style guide" => "Google Python Style Guide"

File: process/Style-guide.md [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+* Limit the maximum number of arguments for function or method to 10
+
+Other things to keep in mind when using Pylint:
+* Use textual pylint overrides e.g. "# pylint: disable=no-self-argument" instead of "# pylint: disable=E0213". For a list of overrides see: http://docs.pylint.org/features.html
+* The canonical pylint configuration for log2timeline project is generated by l2tdevtools. 
+
+
+### Function and method arguments
+
+* Prefix unused function or method arguments with `unused_`.
```
Comment (joachimmetz) [inline process/Style-guide.md]: I think the usage of `_unused` is no longer accurate.

File: process/Style-guide.md [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+# No:
+SomeMethod(True, False)
+
+# Yes:
+SomeMethod(secondary=False)
+```
+
+## Strings
+
+* Quote strings as ' or """ and not "
```
Comment (joachimmetz) [inline process/Style-guide.md]: I opt changing this to:
`Quote strings as ' (1x single quote) or """ (3x double quote) and not " (1x double quote)`

File: process/Style-guide.md [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+  * Use positional or parameter format specifiers with typing e.g. '{0:s}' or '{text:s}' instead of '{0}', '{}' or '{:s}'. If we ever want to have language specific output strings we don't need to change the entire codebase (again). It also makes is easier in determining what type every parameter is expected to be.
+
+## Exceptions
+
+* When catching exceptions use "as exception:" not some alternative form like "as error:" or "as details:"
+* Raise exceptions like this: ```raise MyException('Error message')``` or ```raise MyException```.
+* Although Python allows for ```try ... except ... else``` we prefer not to use it.
+* Make exception messages as useful and descriptive and possible. For example, if an argument is out of an acceptable range, print the invalid value to speed-up debugging.
+
+##Return statements
```
Comment (joachimmetz) [inline process/Style-guide.md]: Change to `## Return statements` (add space)

File: process/Style-guide.md [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+* Use `return` in a function or method that does not return a value.
+
+## Docstrings
+
+* Use English, and avoid abbreviations. Use "for example" or "such as" instead of Latin abbreviations like "e.g.".
+  * Acronyms and initialisms should be preserved, for example HTMLParser and not HtmlParser.
+* We use "Google Style" docstrings. 
+  * For examples, see [this page](http://sphinxcontrib-napoleon.readthedocs.org/en/latest/example_google.html) as well as the notes below. 
+
+There are still a few legacy docstrings in the codebase, here are some examples you might see. Please don't write new code that looks like this:
```
Comment (joachimmetz) [inline process/Style-guide.md]: Is this still accurate?

File: process/Style-guide.md [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+```
+
+### Multiple return types
+
+Python simulates multiple arguments being returned by implicitly returning a tuple. Document like so:
+
+```
+...
+Returns:
+  tuple: containing:
```
Comment (joachimmetz) [inline process/Style-guide.md:155]: Is this still accurate with the pylint docstring check?

File: process/Style-guide.md [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+### Class attributes
+
+In addition to the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html)  sort class attribute alphabetically by name.
+
+```
+class SampleClass(object):
+  """Summary of class here.
+
+  Attributes:
+      eggs (int): number of eggs we have laid.
```
Comment (joachimmetz) [inline process/Style-guide.md]: `-`2 spaces in the indentation

File: process/Style-guide.md [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+
+In addition to the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html) sort instance attribute alphabetically by name inside the constructor (`__init__`).
+
+```
+class SampleClass(object):
+  """Summary of class here."""
+
+  def __init__(self):
+    """Summary of method here."""
+    self.__private_attribute = None
```
Comment (joachimmetz) [inline process/Style-guide.md]: Is `self.__private_attribute` still accurate? Or should we change this? I don't think we use private attributes.

File: process/Style-guide.md [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+## Other
+
+* Avoid the use of global variables.
+* Use class methods in preference to static methods
+* At the top of source files define the encoding, which should be UTF-8, e.g.:
+```
+# -*- coding: utf-8 -*-
+```
+Also see: [PEP 0263](https://www.python.org/dev/peps/pep-0263/)
+
```
Comment (joachimmetz) [inline process/Style-guide.md]: nit: `-` white lines

File: process/Style-guide.md [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+### Constructors
+
+In addition to the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html) sort instance attribute alphabetically by name inside the constructor (`__init__`).
+
+```
+class SampleClass(object):
+  """Summary of class here."""
+
+  def __init__(self):
+    """Summary of method here."""
```
Comment (joachimmetz) [inline process/Style-guide.md:204]: add:
`super(SampleClass, self).__init__()`

**Review by joachimmetz — CHANGES_REQUESTED** [review]
I flagged some parts of which I think they are no longer accurate. Please double check.

File: process/Style-guide.md (hunk 1 above)
Comment (Onager) [reply process/Style-guide.md]: Done

File: process/Style-guide.md (hunk 2 above)
Comment (Onager) [reply process/Style-guide.md]: Still relevant as far I can see...

File: process/Style-guide.md (hunk 3 above)
Comment (Onager) [reply process/Style-guide.md]: Done

File: process/Style-guide.md (hunk 6 above)
Comment (Onager) [reply process/Style-guide.md:155]: Yes.

File: process/Style-guide.md (hunk 5 above)
Comment (Onager) [reply process/Style-guide.md]: Possibly, but I don't think it's useful to have all this extra text. I'll clean it up now.

File: process/Style-guide.md (hunk 7 above)
Comment (Onager) [reply process/Style-guide.md]: Done

File: process/Style-guide.md (hunk 8 above)
Comment (Onager) [reply process/Style-guide.md]: Yeah, we don't use these. Removed these.

File: process/Style-guide.md (hunk 9 above)
Comment (Onager) [reply process/Style-guide.md]: Done

File: process/Style-guide.md (hunk 10 above)
Comment (Onager) [reply process/Style-guide.md:204]: Done

File: process/Style-guide.md (hunk 2 above)
Comment (joachimmetz) [reply process/Style-guide.md]: What I've noticed with the enforcement of matching docstrings Args is that it only applies to args and kwargs (and self and cls)

File: process/Style-guide.md (hunk 4 above)
Comment (joachimmetz) [reply process/Style-guide.md]: You'll need to add a space here otherwise markdown is broken.

**Review by joachimmetz — CHANGES_REQUESTED** [review]
Still one issue not addressed, the one that breaks markdown ;)

File: process/Style-guide.md (hunk 2 above)
Comment (Onager) [reply process/Style-guide.md]: OK, this might be more confusing than useful - removed.

File: process/Style-guide.md (hunk 4 above)
Comment (Onager) [reply process/Style-guide.md]: Woops - fixed.

**Review by joachimmetz — APPROVED** [review]
LGTM

**Pull request merged**